bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -259,7 +259,7 @@ func (r *KubeadmConfigReconciler) reconcile(ctx context.Context, scope *Scope, c
 	}
 	switch {
 	// Wait for the infrastructure to be ready.
-	case !cluster.Status.InfrastructureReady:
+	case cluster.Spec.InfrastructureRef != nil && !cluster.Status.InfrastructureReady:
```

Comment:
I think with the change I suggested on the other PR this change won't be necessary anymore